src/psyclone/psyir/nodes/omp_clauses.py

```diff
@@ -115,3 +117,223 @@ class OMPNogroupClause(Clause):
+        return ""
+
+
+    @staticmethod
+    def _validate_child(position, child):
+        '''
+        Decides whether a given child and position are valid for this node.
+        Any number of Reference nodes are allowed.
+
+        :param int position: the position to be validated.
+        :param child: a child to be validated.
+        :type child: :py:class:`psyclone.psyir.nodes.Node`
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        '''
+        return isinstance(child, Reference)
+
+
+class OMPPrivateClause(Clause):
+    '''
+    OpenMP private clause. This is used to declare variables as private
+    to an OpenMP region.
+    '''
+    _children_valid_format = "[Reference]*"
+    _text_name = "PrivateClause"
+
+    @property
+    def _clause_string(self):
+        if len(self.children) > 0:
+            return "private"
+        return ""
+
+    @staticmethod
+    def _validate_child(position, child):
+        '''
+        Decides whether a given child and position are valid for this node.
+        Any number of Reference nodes are allowed.
+
+        :param int position: the position to be validated.
+        :param child: a child to be validated.
+        :type child: :py:class:`psyclone.psyir.nodes.Node`
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        '''
+        return isinstance(child, Reference)
+
+
+class OMPFirstprivateClause(Clause):
+    '''
+    OpenMP firstprivate clause. This is used to declare variables as
+    firstprivate to an OpenMP region.
+    '''
+    _children_valid_format = "[Reference]*"
+    _text_name = "FirstprivateClause"
+
+    @property
+    def _clause_string(self):
+        if len(self.children) > 0:
+            return "firstprivate"
+        return ""
+
+    @staticmethod
+    def _validate_child(position, child):
+        '''
+        Decides whether a given child and position are valid for this node.
+        Any number of Reference nodes are allowed.
+
+        :param int position: the position to be validated.
+        :param child: a child to be validated.
+        :type child: :py:class:`psyclone.psyir.nodes.Node`
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        '''
+        return isinstance(child, Reference)
+
+
+class OMPDefaultClause(Clause):
+    '''
+    OpenMP Default clause. Used to determine the default declaration for
+    variables used in an OpenMP region.
+    '''
+
+    class DefaultClauseTypes(Enum):
+        '''Enumeration of the different types of OMPDefaultClause supported
+        in PSyclone'''
+        SHARED=0
+        NONE=1
+        FIRSTPRIVATE=2
+
+    _children_valid_format = None
+    _text_name = "OMPDefaultClause"
+
+    @property
+    def _clause_string(self):
+        clause_string = "default("
+        if self._clause_type == OMPDefaultClause.DefaultClauseTypes.SHARED:
```

Comment:
I think there's some way of getting the name of a member of an enumeration so you don't have to do this.

Reply:
I switched this to use name.lower()